### ISSUE TITLE
There is no delete mechanism for attachments

### ISSUE THREAD

**Issue by couling** [go-gitea/gitea#677]
Refering to:
- Gitea version (or commit ref): v1.0.0-126-g302fa42
- Feature Request.

There is no delete mechanism at all for attachments.
- [ ] The user should be presented with the option to delete any / all attachments from an issue comment.
- [x] Attachments should be deleted immediately when an issue / issue comment are deleted.
- [x] Attachments should be deleted immediately when a release is deleted
- [ ] "Orphaned"Attachments should be deleted periodically (eg: cron job).  *Orphaned attachments have both `attachment.CommentID` and `attachment.ReleaseId` of `0`.  They occur when a user starts creating an issue, uploads an attachment and then closes the browser without submitting.*

---
Want to back this issue? [**Post a bounty on it!**](https://www.bountysource.com/issues/40975238-there-is-no-delete-mechanism-for-attachments?utm_campaign=plugin&utm_content=tracker%2F47456670&utm_medium=issues&utm_source=github) We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F47456670&utm_medium=issues&utm_source=github).

**Comment by bkcsoft** [comment]
Also when a release is removed? (since that is now a thing 😄 #673 )

**Comment by couling** [comment]
Actually it looks like the code tries to delete an attachment on deleting a comment.

[See Here](https://github.com/go-gitea/gitea/blob/a6f5efa0bb7bab30b6acd654c5ccf1c273c8b60c/models/issue_comment.go#L119)

However it doesn't manage to do so.

**Comment by couling** [comment]
Okay so [this](https://github.com/go-gitea/gitea/blob/a6f5efa0bb7bab30b6acd654c5ccf1c273c8b60c/models/issue_comment.go#L473) doesn't trigger the `AfterDelete()`.  I guess that's an xorm bug.  I think for the time being I'm going to work around it and explicitly Delete the attachment in `DeleteCommentById`

**Comment by lunny** [comment]
change [this](https://github.com/go-gitea/gitea/blob/a6f5efa0bb7bab30b6acd654c5ccf1c273c8b60c/models/issue_comment.go#L473) to below code, then `AfterDelete()` will work.

```Go
if _, err = sess.Delete(&Comment{
ID: comment.ID,
}); err != nil {
		return err
	}
```

**Comment by appleboy** [comment]
@lunny I will take it.

**Comment by appleboy** [comment]
@lunny we need to merge this XORM PR first https://github.com/go-xorm/xorm/pull/525

**Comment by appleboy** [comment]
#704 also fix item two.

> Attachments should be deleted immediately when an issue / issue comment or release are deleted.

cc @couling

**Comment by bkcsoft** [comment]
These aren't all fixed.

**Comment by ghost** [comment]
The user should be presented with the option to delete any / all previously added attachments while editing a release.

**Comment by ghost** [comment]
@gmastergreatee You are right! In addition to that, you cannot delete and recreate a release because it also deletes the git tag. One should split this.

**Comment by thmo** [comment]
As an additional issue, the attachments remain on disk even after deleting the release itself (see #2618).

**Comment by 6543** [comment]
closed by #14130
